File: fixtures/scale/r/large/core/store.r
# Append-only results file under the working directory.

results_path <- function() {
  path <- Sys.getenv("CIPHBENCH_RESULTS", unset = "")
  if (nzchar(path)) path else "results.csv"
}

store_init <- function() {
  writeLines(character(0), results_path())
}

store_append <- function(row) {
  cat(row, "\n", file = results_path(), append = TRUE, sep = "")
}

File: fixtures/scale/julia/large/src/core/Store.jl
module BenchStore

export store_init, store_append, results_path

function results_path()
    path = get(ENV, "CIPHBENCH_RESULTS", "")
    return isempty(path) ? "results.csv" : path
end

function store_init()
    open(results_path(), "w") do io
        nothing
    end
end

function store_append(row::String)
    open(results_path(), "a") do io
        println(io, row)
    end
end

end
